BitwardenShared/Core/Platform/Services/StateServiceTests.swift

```diff
@@ -1773,6 +1789,64 @@ class StateServiceTests: BitwardenTestCase { // swiftlint:disable:this type_body
         XCTAssertEqual(appSettingsStore.usesKeyConnector["1"], true)
     }
 
+    /// `syncToAuthenticatorPublisher()` returns a publisher for the user's sync to authenticator settings.
+    func test_syncToAuthenticatorPublisher() async throws {
+        await subject.addAccount(.fixture(profile: .fixture(userId: "1")))
+
+        var publishedValues = [SyncToAuthenticatorValue]()
+        let publisher = await subject.syncToAuthenticatorPublisher()
+            .sink(receiveValue: { userId, shouldSync in
+                publishedValues.append(SyncToAuthenticatorValue(userId: userId, shouldSync: shouldSync))
+            })
+        defer { publisher.cancel() }
+
+        try await subject.setSyncToAuthenticator(true)
+
+        XCTAssertEqual(
+            publishedValues,
+            [
+                SyncToAuthenticatorValue(userId: "1", shouldSync: false),
+                SyncToAuthenticatorValue(userId: "1", shouldSync: true),
+            ]
+        )
+    }
+
+    /// `syncToAuthenticatorPublisher()` gets the initial stored value if a cached value doesn't exist.
+    func test_syncToAuthenticatorPublisher_fetchesInitialValue() async throws {
+        await subject.addAccount(.fixture(profile: .fixture(userId: "1")))
+
+        appSettingsStore.syncToAuthenticatorByUserId["1"] = true
+
+        var publishedValues = [SyncToAuthenticatorValue]()
+        let publisher = await subject.syncToAuthenticatorPublisher()
+            .sink(receiveValue: { userId, shouldSync in
+                publishedValues.append(SyncToAuthenticatorValue(userId: userId, shouldSync: shouldSync))
+            })
+        defer { publisher.cancel() }
+
+        try await subject.setSyncToAuthenticator(false)
+
+        XCTAssertEqual(
+            publishedValues,
+            [
+                SyncToAuthenticatorValue(userId: "1", shouldSync: true),
+                SyncToAuthenticatorValue(userId: "1", shouldSync: false),
+            ]
+        )
+    }
+
+    /// `syncToAuthenticatorPublisher()` returns false if the user is not logged in.
+    func test_syncToAuthenticatorPublisher_notLoggedIn() async throws {
+        var publishedValues = [SyncToAuthenticatorValue]()
+        let publisher = await subject.syncToAuthenticatorPublisher()
+            .sink(receiveValue: { userId, shouldSync in
+                publishedValues.append(SyncToAuthenticatorValue(userId: userId, shouldSync: shouldSync))
+            })
+        defer { publisher.cancel() }
```

Comment:
These defers are nice. I've been putting them at the end for publishers and Tasks in tests, but I'm going to change to defers :)